### ISSUE TITLE
timezone detection not reliable

### ISSUE THREAD

**Issue by nbriggs** [Interlisp/medley#65]
Timezone detection doesn't appear to work on Linux systems, but does work on MacOS.

**Comment by masinter** [comment]
reopen to add documentation and setting the TZ environment variable in the run-medley scripts

**Comment by nbriggs** [comment]
Please DO NOT set the TZ variable in the script.
That will override the information from the /etc/localtime and it will make it harder for the user to override the default system setting, if they want, by setting TZ themselves.   Since this is a normal part of the system, and all the systems we're trying to support should just work now, I don't know that it needs much in the way of documentation.

If you want to open a documentation issue to mention it, that would be appropriate.

**Comment by masinter** [comment]
TZ isn't set on WSL or docker configurations. I think maybe WHO-LINE has a 'set time" option that does nothing now and could be repurposed to set the time-zone.

**Comment by nbriggs** [comment]
What about /etc/localtime ?

**Comment by nbriggs** [comment]
I think that starting up the Docker image should by default run it in the timezone of the host, but could have an option to override that -- but it should be at the system level so that the whole docker image has the right idea of the time.  I can imagine getting burned by per-process timezones that are not consistent with the kernel (and thus file system).

I think Medley/maiko/lde should get its timezone from the environment.

**Comment by masinter** [comment]
what do I need to do to get my who-line time to show the correct local time?
I pulled the latest maiko, made new lde, ran it, and ... still showing utc time

**Comment by nbriggs** [comment]
Running on what OS?

**Comment by nbriggs** [comment]
If you're trying to set the timezone in WSL2, first look at whether /etc/localtime exists and if it's a link, what it's a link to.  Also check if /etc/timezone exists.

If /etc/timezone doesn't exist, and /etc/localtime does then try:
```
sudo mv /etc/localtime /etc/localtime.orig
sudo ln -s /usr/share/zoneinfo/America/Los_Angeles /etc/localtime
```
If /etc/timezone exists, and you have dpkg-reconfigure then do
```
sudo dpkg-reconfigure tzdata
```
and answer the questions.  This should set up /etc/timezone.
Check that the "date" command in the WSL shell produces the correct date/time.

**Comment by masinter** [comment]
works great now, sorry for the confusion
no documentation or init file etc. needed

**Comment by nbriggs** [comment]
No worries.  Which fix worked?

**Comment by nbriggs** [comment]
FreeBSD does not support the POSIX extern timezone method for determining the timezone offset.

**Comment by nbriggs** [comment]
FreeBSD also doesn't appear to return non-zero timezone information in the gettimeofday() results, however the tm_gmtoff value from a call to localtime() DOES contain the correct offset, so use that on FreeBSD.

**Comment by nbriggs** [comment]
Fixed with maiko@d8c0ab9bdf6d9ad5116851014a9bc782baa8f313
